### ISSUE TITLE
GHGist returns an unexpected id

### ISSUE THREAD

**Issue by Aulig** [github-api/github-api#682]
GHGist.getId() returns a long id, however the expected value would be the String id contained in GHGist. This field is only accessible via reflection though.

**Comment by bitwiseman** [comment]
@Aulig  What field is that?

**Comment by Aulig** [comment]
Huh, Im a bit confused now, as its not defined in the actual source code, I just looked at the source with the intellij source viewer before. I guess the field is generated afterwards?

Either way, it occurs in the equals() method of GHGist:
return id.equals(ghGist.id);

This is the actual string id of the gist.

**Comment by bitwiseman** [comment]
@Aulig  and what should it be instead? Here's the github api doc for comparison. 
https://developer.github.com/v3/gists/

**Comment by Aulig** [comment]
There should be a way to get this id. Currently there is a GHGist.getId() function, however it is from a superclass and returns a long value and not this string.

**Comment by bitwiseman** [comment]
Ah, so this line looks like it defines `id`: 
https://github.com/github-api/github-api/blob/14f3660f55058e81173ed81bdb94ce40e56c250c/src/main/java/org/kohsuke/github/GHGist.java#L26


But this is what gets call: 
https://github.com/github-api/github-api/blob/14f3660f55058e81173ed81bdb94ce40e56c250c/src/main/java/org/kohsuke/github/GHObject.java#L107-L109

And it references the `int id` defined in that class. 😭

**Comment by Aulig** [comment]
Yes exactly, I mustve missed the line defining id earlier because I'm on my phone, sorry about that!

**Comment by bitwiseman** [comment]
Fixing this is not as easy as it sounds.  The return type of getId is a long but the id from GHGist is not always a `long`.  😭

**Comment by Aulig** [comment]
Yea the GHGist ie is usually a string. I would've simply added a new function like 'getGistId()' that returns this String. However I'm aware how this is not an optimal solution, but I guess it's better than nothing. I assume most people use an IDE with auto-completion like me, so I just did GHGist.getId(), but if I saw that it also suggested another method GHGist.getGistId() I wouldve tried them both or read their documentation, which wouldve avoided the hassle I went through originally :D
